fuel-tx/src/transaction/validity.rs

```diff
@@ -317,23 +312,49 @@ where
 pub(crate) fn check_common_part<T>(
     tx: &T,
     block_height: BlockHeight,
-    tx_params: &TxParameters,
-    predicate_params: &PredicateParameters,
-    base_asset_id: &AssetId,
+    consensus_params: &ConsensusParameters,
 ) -> Result<(), CheckError>
 where
-    T: field::GasPrice
-        + field::GasLimit
-        + field::Maturity
-        + field::Inputs
-        + field::Outputs
-        + field::Witnesses,
+    T: canonical::Serialize + Chargeable + field::Outputs,
 {
-    if tx.gas_limit() > &tx_params.max_gas_per_tx {
-        Err(CheckError::TransactionGasLimit)?
+    let ConsensusParameters {
+        tx_params,
+        predicate_params,
+        base_asset_id,
+        gas_costs,
+        fee_params,
+        ..
+    } = consensus_params;
+
+    check_size(tx, tx_params)?;
+
+    if !tx.policies().is_valid() {
+        Err(CheckError::TransactionPoliciesAreInvalid)?
+    }
+
+    if tx.policies().get(PolicyType::GasPrice).is_none() {
+        Err(CheckError::TransactionNoGasPricePolicy)?
+    }
+
+    if let Some(witness_limit) = tx.policies().get(PolicyType::WitnessLimit) {
+        let witness_size = tx.witnesses().size_dynamic();
+        if witness_size as u64 > witness_limit {
+            Err(CheckError::TransactionWitnessLimitExceeded)?
+        }
+    }
+
+    let max_gas = tx.max_gas(gas_costs, fee_params);
+    if max_gas > tx_params.max_gas_per_tx {
+        Err(CheckError::TransactionMaxGasExceeded)?
+    }
```

Comment:
The specification says `GasLimit > MAX_GAS_PER_TX`, while in real life should limit the total transaction cost.

I've created a [specification PR](https://github.com/FuelLabs/fuel-specs/pull/533) to actualize it.

But it is a huge breaking change because SDK right now uses `MAX_GAS_PER_TX ` as a `GasLimit` -> the transaction always fails because of this check.